### ISSUE TITLE
The headers do not render correctly when the whole hierarchy is not respected.

### ISSUE THREAD

**Issue by EKliptus** [Aidurber/obsidian-plugin-dynamic-toc#38]
Yes it is very bad to skip a level of heading !
But if I can, I do it ! And the official obsidian plan view render it properly...
Example, for this document : 
```
## Level 1
Bal bla bla
#### Level 3
Bla bla bla too
```
Note that the `### Level 2` heading is skipped.

The dynamic toc render this : 
`o Level 1 - Level 3`

Thanks for your plugin, it's very useful.

**Comment by Aidurber** [comment]
I think I understand you and I can reproduce it. 

![image](https://user-images.githubusercontent.com/5732291/151693352-bb208fb4-73e0-48f7-9f92-f2ed16242714.png)

I'll try and get around to this tomorrow.

**Comment by Aidurber** [comment]
# Problem

Okay so I dug into this issue and the problem is that it cannot produce valid markdown if the heading levels have more than 1 level between the parent and the child. 

What I mean by this is a level 2 heading followed by a level 4 heading. 
```md
## Level 2
#### Level 4
Notice the extra two #
```

Opposed to treating headers as they should be (offering lexical meaning to a document, see #26)
```md
## Level 2
### Level 3
```

# Solution
I managed to come up with a solution but I don't recommend it. I've added a new setting "Allow inconsistent heading levels" you can turn this on and it will support your use case. 

With a layout like this: 
![2022-02-01 08_25_01-Debug - Obsidian v0 13 23](https://user-images.githubusercontent.com/5732291/151935626-73ae1c68-34b1-4829-96c6-8c74a30f6405.jpg)

It will produce a table like this: 
![2022-02-01 08_25_12-Debug - Obsidian v0 13 23](https://user-images.githubusercontent.com/5732291/151935627-9951e5be-a492-47ca-b259-ba5adf37cc3f.jpg)

Notice how the level 3 and level 4 headings are at the same indentation level. This is the only way to support these heading depths because we cannot generate valid markdown otherwise. 

This is available in version 0.0.24. Please let me know if it works for your use case.
